crates/noirc_frontend/src/monomorphisation/mod.rs

```diff
@@ -129,49 +129,12 @@ impl Monomorphiser {
         self.globals.entry(id).or_default().insert(typ, new_id);
     }
 
-    /// The main function is special, we need to check for a return type and if present,
```

Comment:
@jfecher This is the main part which requires your suggestions.

We no longer need to apply this constrain and instead we process the return Operation in SSA which only happens on main. Processing involves getting the Witness indices and putting them into the `public_inputs` vector in ACIR